### PR TITLE
docs: Flank-scripts command overview

### LINKED ISSUE #1545: Redesign flank-scripts commands structure and namings
**Author the user story for this feature**

As a flank-scripts user, I want to have them well-structured and described so I can clearly understand the purpose of each

**Describe the solution**

* Review flank-scripts commands from a new developer perspective. 
* Flatten unneeded nestings. 
* Make sure the commands are grouped in a reasonable way.
* Improve descriptions if needed

### PR DESCRIPTION
**Pull request by piotradamczyk5**
Fixes #1545

This PR contains documentation for the proposition of restructuring flank-scripts CLI commands as well as a basic description of each of it

### CODE REVIEW

File: docs/flank-scripts/command_overview.md [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+___
+
+- `integration_tests` - Group of commands for handling integration tests (1)
+    - `process_results` - Process results of integration tests
+
+
+(1) - currently there is only one command, but probably there will be more in future
+___
+    
+- `copy_issue_properties` - Copy properties(assignees, SP, labels) from issue to PR
```
Comment (jan-gogo) [inline docs/flank-scripts/command_overview.md:33]: fix - name

File: docs/flank-scripts/command_overview.md [hunk 2]
```diff
@@ -0,0 +1,110 @@
+# Flank-scripts command overview
+
+### Command List
+
+- `release` - Group of commands for creating Flank release
```
Comment (jan-gogo) [inline docs/flank-scripts/command_overview.md:5]: I am wondering if most of the commands in this scope are more related to `github` than the `release`. For example, if we would have similar commands for a different provider that would be more convenient to add a new group for it and reuse the common part like `generate_release_notes`.

Comment (piotradamczyk5) [reply docs/flank-scripts/command_overview.md:5]: Decided to make a separate package for `github`

File: docs/flank-scripts/command_overview.md [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+
+___
+    
+- `coding_style` - Group of commands used for applying correct coding style
+    - `apply_linter_to_git_hooks` - Apply KtLint pre-commit hook
+    - `apply_linter_to_ide` - Apply KtLint to IDE
+  
+___
+    
+- `update_dependencies` - Update repository dependencies
```
[the remark this answers is not in the document]
Comment (piotradamczyk5) [reply docs/flank-scripts/command_overview.md:22]: Decided to move to a new package `github`

File: docs/flank-scripts/command_overview.md (hunk 1 above)
Comment (piotradamczyk5) [reply docs/flank-scripts/command_overview.md:33]: Will be moved to new `github`

File: docs/flank-scripts/command_overview.md [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+___
+    
+- `assemble` - Group of commands for assemble test application
+    - `android_app` - Assemble Android test application
+    - `go_artifacts` - Generate go artifacts
+    - `ios_earl_grey` - Assemble iOS earl grey application
+    - `ios_flank_example` - Assemble iOS flank example application
+    - `ios_game_loop` - Assemble iOS game loop application
+    - `ios_test_plans` - Assemble iOS test plans application
+    - `ios_all` - Assemble all iOS applications
```
Comment (piotradamczyk5) [inline docs/flank-scripts/command_overview.md:63]: make structure assemble [ios|android] <command>

File: docs/flank-scripts/command_overview.md [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+    - `delete_snapshot` - Delete snapshot package from artifacts repository
+    - `delete_old_tag` - Delete old/unused tag from Github
+    - `sync_with_maven_central` - Sync artifacts repository with Maven central
+    - `make_github_release` - Make new Github release
+
+___
+    
+- `coding_style` - Group of commands used for applying correct coding style
+    - `apply_linter_to_git_hooks` - Apply KtLint pre-commit hook
+    - `apply_linter_to_ide` - Apply KtLint to IDE
```
[the remark this answers is not in the document]
Comment (piotradamczyk5) [reply docs/flank-scripts/command_overview.md]: fixed

File: docs/flank-scripts/command_overview.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (piotradamczyk5) [reply docs/flank-scripts/command_overview.md]: fixed

File: docs/flank-scripts/command_overview.md [hunk 6]
```diff
@@ -0,0 +1,110 @@
+# Flank-scripts command overview
+
+### Command List
+
+- `release` - Group of commands for creating Flank release
+    - `generate_release_notes` - Generate release notes
+    - `next_tag` - Get tag for next release
```
[the remark this answers is not in the document]
Comment (piotradamczyk5) [reply docs/flank-scripts/command_overview.md]: fixed

File: docs/flank-scripts/command_overview.md (hunk 1 above)
Comment (piotradamczyk5) [reply docs/flank-scripts/command_overview.md]: fixed

File: docs/flank-scripts/command_overview.md [hunk 7]
```diff
@@ -0,0 +1,119 @@
+# Flank-scripts command overview
+
+### Command List
+
+- `assemble` - Group of commands for assemble test application
```
Comment (Sloox) [inline docs/flank-scripts/command_overview.md:5]:
```suggestion
- `assemble` - Group of commands to assemble test applications
```

File: docs/flank-scripts/command_overview.md [hunk 8]
```diff
@@ -0,0 +1,119 @@
...
+    - `ios` - Subgroup of commands for managing assemble iOS apps
```
Comment (Sloox) [inline docs/flank-scripts/command_overview.md:6]:
```suggestion
    - `ios` - Subgroup of commands for iOS app assembly
```

File: docs/flank-scripts/command_overview.md [hunk 9]
```diff
@@ -0,0 +1,119 @@
...
+        - `earl_grey` - Assemble iOS earl grey application
+        - `flank_example` - Assemble iOS flank example application
+        - `game_loop` - Assemble iOS game loop application
+        - `test_plans` - Assemble iOS test plans application
+        - `all` - Assemble all iOS applications
+    - `android` - Subgroup of command for managing assemble Android apps
```
Comment (Sloox) [inline docs/flank-scripts/command_overview.md:12]:
```suggestion
    - `android` - Subgroup of command for Android app assembly
```

File: docs/flank-scripts/command_overview.md [hunk 10]
```diff
@@ -0,0 +1,119 @@
...
+        - `app` - Assemble Android test application
+    - `go_artifacts` - Generate go artifacts
+    
+___
+
+- `firebase` - Group of commands for managing firebase integrations
+    - `update_api` - Update api schema
+    - `generate_client` - Generate Java Client based on api schema
+    - `check_for_sdk_updates` - Check for new SDK features and create tasks for it
```
Comment (Sloox) [inline docs/flank-scripts/command_overview.md:21]:
```suggestion
    - `check_for_sdk_updates` - Check for new SDK features and create update tasks for it
```

File: docs/flank-scripts/command_overview.md [hunk 11]
```diff
@@ -0,0 +1,119 @@
...
+    - `link` - Create symbolic link to under test_runner/src/test/kotlin/ftl/fixtures/tmp to test_artifacts/{branchName}.
+    - `prepare`- Creates fresh copy of test artifacts for current working branch, basing on existing one.
+    - `remove_remote` - Remove remote copy of test artifacts.
+    - `resolve` - Automatically prepare local artifacts if needed.
+    - `unzip` - Unpack test artifacts zip archive.
+    - `upload` - Upload test artifacts zip as github release asset.
+    - `zip` - Create zip archive from test artifacts directory.
+
+  
+(1) - currently there is only one command, but probably there will be more in future
```
Comment (Sloox) [inline docs/flank-scripts/command_overview.md:68]:
```suggestion
(1) - denotes that there is only one command, but it may change in the future.
```

**Review by pawelpasterz — APPROVED** [review]

**Review by axelzuziak-gogo — APPROVED** [review]

**Pull request merged**